src/pages/create_index/states/define_weights/index.tsx

```diff
@@ -0,0 +1,151 @@
+import { IconButton, Slider, TextField } from '@material-ui/core'
+import { BackButton, PrimaryButton } from 'components/buttons'
+import { CreateIndexStates, SelectCurrenciesProps } from 'pages/create_index'
+import React, { useState } from 'react'
+import { useStyles } from './styles'
+import DeleteIcon from 'assets/icons/DeleteIcon.svg'
+
+const DefineWeightsState = ({
+  setPageState,
+  selectedCurrencies,
+  setSelectedCurrencies,
+}: {
+  setPageState: React.Dispatch<React.SetStateAction<CreateIndexStates>>
+  selectedCurrencies: SelectCurrenciesProps[]
+  setSelectedCurrencies: React.Dispatch<React.SetStateAction<SelectCurrenciesProps[]>>
+}) => {
+  const classes = useStyles()
+
+  const filteredArr = selectedCurrencies.filter((item) => item.selected)
+  const fullPercentage = 100
+
+  interface ValuesProps {
+    id: number
+    value: number
+  }
+
+  const testArr = filteredArr.map((item) => {
+    return { id: item.id, value: 0 }
+  })
+
+  const testValues = testArr as ValuesProps[]
+
+  const [values, setValues] = useState<ValuesProps[]>(testValues)
+
+  const handleSliderChange = (newValue: number, index: number) => {
+    const initialState = values.map((obj: ValuesProps) => (obj.id === index ? { ...obj, value: newValue } : obj))
+
+    const allValues = initialState
+      .map((el) => el.value)
+      .reduce((accumulator, currentValue) => {
+        return accumulator + currentValue
+      })
+    const newState = values.map((obj: ValuesProps) =>
+      obj.id === index
+        ? { ...obj, value: newValue >= 100 ? 100 : newValue }
+        : allValues > 100
+        ? {
+            ...obj,
+            value: obj.value > 0 ? obj.value - Math.ceil((allValues - 100) / (values.length - 1)) : 0,
+          }
+        : obj,
+    )
+
+    setValues(newState)
+  }
+
+  const handleDelete = (id: number) => {
+    const newState = selectedCurrencies.map((obj: SelectCurrenciesProps) =>
+      obj.id === id ? { ...obj, selected: !obj.selected } : obj,
+    )
+    setSelectedCurrencies(newState)
+  }
+
+  const handleNavigate = () => {
+    const newState = selectedCurrencies.map((obj: SelectCurrenciesProps) => {
+      const currentWeight =
```

Comment:
try this const currentWeight = values.find((el) => el.id === obj.id)?.value ?? 0